scripts/markdown_seo_lint.js

```diff
@@ -0,0 +1,219 @@
+                // Include the error message in the front matter error object
+                // so we can display it to the user.
+                result.frontMatter[fullPath] = {
+                    error: e.message,
+                };
+                result.files.push(fullPath);
+            }
+        }
+
+        // If there are remaining paths in the list, keep going.
+        const remaining = paths.slice(1, paths.length);
+        if (remaining.length > 0) {
+            return searchForMarkdown(remaining, result);
+        }
+    }
+
+    return result;
+}
+
+// Get a list of markdown files.
+function getMarkdownFileList(parentPath) {
+    const fullParentPath = path.resolve(__dirname, parentPath)
+    const dirs = fs.readdirSync(fullParentPath).map(function(dir) {
+        return parentPath + "/" + dir;
+    });
+
+    return searchForMarkdown(dirs);
+}
+
+const filesToLint = getMarkdownFileList("../content");
+
+const opts = {
+    files: filesToLint.files,
+    config: {
+        // Allow inline HTML.
+        MD033: false,
+        // Do not enforce line length.
+        MD013: false,
+        // Don't force language specification on code blocks.
+        MD040: false,
+        // Allow hard tabs.
+        MD010: false,
+        // Allow puncuation in headers.
+        MD026: false,
+        // Allow dollars signs in code blocks without values
+        // immediately below the command.
+        MD014: false,
+        // Allow all code block styles in a file. Code block styles
+        // are created equal and we shall not discriminate.
+        MD046: false,
+        // Allow indents on unordered lists to be 4 spaces instead of 2.
+        MD007: { indent: 4 },
+        // Allow duplicate heading content inside child headings.
+        MD024: { siblings_only: true },
+    },
+};
+
+const result = markdownlint.sync(opts);
+
+// Get the total number of errors and build a collection
+// of lint errors.
+const errorsCount = [].concat.apply([], Object.values(result)).length;
+const errors = Object.keys(result).map(function(key) {
+    const lintErrors = result[key];
+    const frontMatterErrors = filesToLint.frontMatter[key];
+
+    if (frontMatterErrors.error) {
+        lintErrors.push({
+            lineNumber: "File Header",
+            ruleDescription: frontMatterErrors.error,
+        });
+    }else {
+        if (frontMatterErrors.title) {
+            lintErrors.push({
+                lineNumber: "File Header",
+                ruleDescription: frontMatterErrors.title,
+            });
+        }
+        if (frontMatterErrors.metaDescription) {
+            lintErrors.push({
+                lineNumber: "File Header",
+                ruleDescription: frontMatterErrors.metaDescription,
+            });
+        }
+    }
+
+    if (lintErrors.length > 0) {
+        return { path: key, errors: lintErrors };
+    }else {
+        return null;
+    }
+}).filter(function(item) { return item !== null; });
```

Comment:
As much as I love functional programming, I'd suggest you break this up into multiple expressions for readability. As-is it's a bit too dense.
```
const errorsCount = [].concat.apply([], Object.values(result)).length;
const errors = Object.keys(result).map(function(key) {
    ...
}).filter(function(item) { return item !== null; });
```

Can we split this up into multiple local variables? Or use a function defined somewhere else instead of an anonymous function/lambda?

Ultimately if the "main" part of the script were shorter, things would be a lot easier to follow. e.g.:

```
let errors = groupLintErrorOutput(...);
```